recipes/gcc/all/conanfile.py

```diff
@@ -1,126 +1,193 @@
-                  "libraries for these languages (libstdc++,...). "
+    description = (
+        "The GNU Compiler Collection includes front ends for C, "
+        "C++, Objective-C, Fortran, Ada, Go, and D, as well as "
+        "libraries for these languages (libstdc++,...). "
+    )
     topics = ("gcc", "gnu", "compiler", "c", "c++")
     homepage = "https://gcc.gnu.org"
     url = "https://github.com/conan-io/conan-center-index"
     license = "GPL-3.0-only"
     settings = "os", "compiler", "arch", "build_type"
-    _autotools = None
 
-    def build_requirements(self):
-        self.build_requires("flex/2.6.4")
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self)
-        pkgversion = 'conan GCC %s' % self.version
-        bugurl = self.url + '/issues'
-        libdir = "%s/lib/gcc/%s" % (self.package_folder, self.version)
-        args = [
-            "--enable-languages=c,c++",
-            "--disable-nls",
-            "--disable-multilib",
-            "--disable-bootstrap",
-            "--with-system-zlib",
-            "--with-gmp=%s" % self.deps_cpp_info['gmp'].rootpath,
-            '--with-mpc=%s' % self.deps_cpp_info["mpc"].rootpath,
-            "--with-mpfr=%s" % self.deps_cpp_info["mpfr"].rootpath,
-            "--without-isl",
-            "--libdir=%s" % libdir,
-            '--with-pkgversion=%s' % pkgversion,
-            "--program-suffix=-%s" % self.version,
-            "--with-bugurl=%s" % bugurl
-        ]
-        if self.settings.os == "Macos":
-            xcrun = tools.XCRun(self.settings)
-            args.extend([
-                '--with-native-system-header-dir=/usr/include',
-                "--with-sysroot={}".format(xcrun.sdk_path)
-            ])
-        self._autotools.libs = []  # otherwise causes config.log to fail finding -lmpc
+    def configure(self):
         if self.settings.compiler in ["clang", "apple-clang"]:
-            # xgcc: error: unrecognized command-line option -stdlib=libc++
-            if self.settings.compiler.libcxx == "libc++":
-                self._autotools.cxx_flags.remove("-stdlib=libc++")
-            elif self.settings.compiler.libcxx in ["libstdc++", "libstdc++11"]:
-                self._autotools.cxx_flags.remove("-stdlib=libstdc++")
-        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
-        return self._autotools
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+            # Can't remove this from cxxflags with autotools - so get rid of it
+            del self.settings.compiler.libcxx
+
+    def build_requirements(self):
+        if self.info.settings.os == "Linux":
+            # binutils recipe is broken for Macos, and Windows uses tools
+            # distributed with msys/mingw
+            self.tool_requires("binutils/2.38")
```

Comment:
I'm unclear on what is driving the need to specify binutils as a build-time dependency for Linux. I was able to build fine without it. If there are conditions where it is needed, it would be nice to scope the dependency down to those circumstances. Also, the latest version of Conan 2.0 doesn't like the use of self.info.settings in the build_requirements method, so while my recommendation would be to eliminate this if-block altogether, at a minimum it likely needs to be changed to:
 
> if self.settings.os == "Linux":

Reply:
You were able to build without it because there will be autodiscovery of the your system binutils, which should exist on all linux systems. This is here in the interest of explicitly modelling dependencies and moving away from a reliance on/contamination by your system executables. Specifically, binutils provides the assembler and linker for linux systems. On macos, the assembler and linker are provided by cctools (https://github.com/conan-io/conan-center-index/issues/14169). I think on windows these are bundled with the msys/mingw subsystems.

This allows a user to specify/override the version of binutils being used explicitly to control the binaries used independently of what's installed on your system.